packages/ceramic-core/src/pin/pinning-service-factory.ts

```diff
@@ -0,0 +1,34 @@
+import * as Ipfs from "ipfs"
+
+import { CeramicConfig } from "../ceramic"
+
+import PinningService from "./pinning-service"
+import MockPinningService from "./mock/mock-pinning-service"
+import LevelPinningService from "./level/level-pinning-service"
+
+/**
+ * PinningService factory class
+ */
+export default class PinningServiceFactory {
+    private readonly mockPinningService: MockPinningService
+    private readonly levelPinningService: LevelPinningService
```

Comment:
General thought about the factory pattern (used here and for the anchor service). It seems like it might add more code bloat than what we actually get from it? In general I'm quite concerned with the size of our libraries and would like to keep the code base as minimal as possible while still being very readable. In the example here it seems like it would be quite straightforward to just construct the "pinningService" when we need it.

Related note: I think the name "pinning service" might be confusing since this is an internal interface in the ceramic-core codebase. In the future we will have actual pinning services that pin data as a service in the network. This might cause unecessary confusing. We should consider renaming this to something like "StateStore" or something similar that points to this being a local persistance layer.